HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,44 @@
+def update_dist(level, neighbors_at_level):
+    '''
+    Update Distances of all nodes given a current level and neighbors at next level
+    '''
+    return lambda (node, (dist, neighbors)): (node, (min(dist,level+1), neighbors)) if node in neighbors_at_level else (node, (dist, neighbors))
```

Comment:
You could have used spark's functions, such as join, subtractByKey, and union, to do this (instead of python).